### PR TITLE
Add test to prove that encoding is unchanged after tree update

### PR DESCRIPTION
**Pull request by mattwar**
@Pilchie  @jasonmalinowski please review

### CODE REVIEW

**Comment by jasonmalinowski** [comment]
:+1:

File: src/Workspaces/CoreTest/SolutionTests/SolutionTests.cs [hunk 1]
```diff
@@ -1343,5 +1344,23 @@ public class C : A {
...
+
+        [WorkItem(1088127, "DevDiv")]
+        [Fact]
+        public void TestEncodingRetainedAfterTreeChanged()
+        {
+            var ws = new AdhocWorkspace();
+            var proj = ws.AddProject("proj", LanguageNames.CSharp);
+            var doc = ws.AddDocument(proj.Id, "a.cs", SourceText.From("public class c { }", Encoding.UTF32));
+
+            Assert.Equal(Encoding.UTF32, doc.GetTextAsync().Result.Encoding);
```
Comment (Pilchie) [inline src/Workspaces/CoreTest/SolutionTests/SolutionTests.cs]: I'm pretty sure @KevinH-MS found that these properties aren't actually singletons, so our other tests compare the names :-/

**Comment by Pilchie** [comment]
LGTM apart from the possible identity issue.

Maybe a comment linking this to http://source.roslyn.io/#Roslyn.Services.UnitTests/WorkspaceTests/MSBuildWorkspaceTests.cs,c7fc5227a96eb410 would be nice as well?

File: src/Workspaces/CoreTest/SolutionTests/SolutionTests.cs (hunk 1 above)
Comment (mattwar) [inline src/Workspaces/CoreTest/SolutionTests/SolutionTests.cs]: Its okay to use compare with .Equals, so Assert.Equal is okay but Assert.Same is not.

**Pull request merged**